src/tinypeel/Peeling/PeelingIO.py

```diff
@@ -85,40 +85,58 @@ def writeGenotypes(pedigree, genoProbFunc, isSexChrom):
         )
     if args.geno_prob:
         writeGenoProbs(pedigree, genoProbFunc, args.out_file + ".geno_prob.txt")
-    if args.geno_threshold and args.geno:
-        for thresh in args.geno_threshold:
-            if thresh < 1 / 3:
-                thresh = 1 / 3
-            if args.binary_call_file:
-                writeBinaryCalledGenotypes(
-                    pedigree,
-                    genoProbFunc,
-                    isSexChrom,
-                    args.out_file + ".called." + str(thresh),
-                    thresh,
-                )
-            else:
-                writeCalledGenotypes(
-                    pedigree,
-                    genoProbFunc,
-                    isSexChrom,
-                    args.out_file + ".geno_" + str(thresh) + ".txt",
-                    thresh,
-                )
-
-    if args.hap_threshold and args.hap:
-        for thresh in args.hap_threshold:
-            if thresh < 1 / 2:
-                thresh = 1 / 2
-            if args.binary_call_file:
-                pass  # this function is not applied
-            else:
-                writeCalledPhase(
-                    pedigree,
-                    genoProbFunc,
-                    args.out_file + ".hap_" + str(thresh) + ".txt",
-                    thresh,
-                )
+    if args.geno:
+        if args.geno_threshold:
+            for thresh in args.geno_threshold:
+                if thresh < 1 / 3:
+                    thresh = 1 / 3
+                if args.binary_call_file:
+                    writeBinaryCalledGenotypes(
+                        pedigree,
+                        genoProbFunc,
+                        isSexChrom,
+                        args.out_file + ".called." + str(thresh),
+                        thresh,
+                    )
+                else:
+                    writeCalledGenotypes(
+                        pedigree,
+                        genoProbFunc,
+                        isSexChrom,
+                        args.out_file + ".geno_" + str(thresh) + ".txt",
+                        thresh,
+                    )
+        else:
+            thresh = 1 / 3
+            writeCalledGenotypes(
```

Comment:
if args.geno:
         if args.geno_threshold:
             for thresh in args.geno_threshold:
                 if thresh < 1 / 3:
                     thresh = 1 / 3
         else:
             thresh = 1 / 3

         if args.binary_call_file:
             writeBinaryCalledGenotypes(
                pedigree,
                genoProbFunc,
                isSexChrom,
                args.out_file + ".called." + str(thresh),
                thresh,
             )
         else:
            writeCalledGenotypes(
  ...
```

Do you see what I have done here?

Reply:
```
if args.geno:
 TODO: improve this top part
         treshs = [] 
         if len(args.geno_threshold) > 0:
             for thresh in args.geno_threshold:
                 if thresh < 1 / 3:
                     threshs.append(1 / 3)
         else:
             threshs = 1 / 3

        for tresh in treshs:
           if args.binary_call_file:
               writeBinaryCalledGenotypes(
                  pedigree,
                  genoProbFunc,
                  isSexChrom,
                  args.out_file + ".called." + str(thresh),
                  thresh,
               )
           else:
              writeCalledGenotypes(
  ...
```